byron-proxy/src/exec/Main.hs

```diff
@@ -401,15 +430,90 @@ runClient tracer clientOptions genesisConfig epochSlots db = case clientOptions
   Nothing -> pure ()
 
   Just (Left shelleyClientOptions) -> do
-    addrInfos <- Network.getAddrInfo (Just addrInfoHints) (Just host) (Just port)
-    case addrInfos of
-      [] -> error "no getAddrInfo"
-      (addrInfo : _) -> IPC.runClient addrInfo stringTracer epochSlots db
+    addrInfosLocal  <- Network.getAddrInfo (Just addrInfoHints) (Just "127.0.0.1") (Just "0")
+    addrInfosRemote <- Network.getAddrInfo (Just addrInfoHints) (Just host) (Just port)
+    case (addrInfosLocal, addrInfosRemote) of
+      (addrInfoLocal : _, addrInfoRemote : _) -> connectTo
+        encodeTerm
+        decodeTerm
+        (initiatorVersions epochSlots chainSyncClient)
+        addrInfoLocal
+        addrInfoRemote
+      _ -> error "no getAddrInfo"
     where
 
     host = scoHostName shelleyClientOptions
     port = scoServiceName shelleyClientOptions
     addrInfoHints = Network.defaultHints
+    -- | This chain sync client will first try to improve the read pointer to
+    -- the tip of the database, and then will roll forward forever, stopping
+    -- if there is a roll-back.
```

Comment:
The client bellow is not stopping on a rollback, it just ignores it and goes forward.

Reply:
Yes that's true. It _would_ stop eventually though, because the next database write after a proper roll back would fail.

Reply:
I see, so the DB is making some form of validation that the next thing is the expected one.  

It might be good to this to the a comment explaining this.

Reply:
Sort of, but it's not exactly validation. The DB will give an exception if you try to write a block to a slot in the past (i.e. it has a block for that slot or a greater slot). I'll just make this client fail if a rollback is seen.

Reply:
ok, it makes sense for an immutable data store.